src/QueryPipeline/QueryPipelineBuilder.cpp

```diff
@@ -323,7 +324,15 @@ std::unique_ptr<QueryPipelineBuilder> QueryPipelineBuilder::joinPipelines(
     right->pipe.dropExtremes();
 
     left->pipe.collected_processors = collected_processors;
-    right->pipe.collected_processors = collected_processors;
+    IQueryPlanStep* step = right->pipe.processors.back()->getQueryPlanStep();
+    if (ExpressionStep::name == step->getName())
```

Comment:
isn't `typeid_cast` enough to check that the `step` is `ExpressionStep`?

Reply:
Ah, `typeid_cast` is a good point, thanks.